sdk/metricsadvisor/azure-ai-metricsadvisor/src/main/java/com/azure/ai/metricsadvisor/MetricsAdvisorClient.java

```diff
@@ -193,7 +193,7 @@
      *
      * @return the list of enrichment status's for the specified metric.
      * @throws IllegalArgumentException thrown if {@code metricId} fail the UUID format validation.
-     * @throws ErrorCodeException thrown if the request is rejected by server.
+     * @throws HttpResponseException thrown if the request is rejected by server.
```

Comment:
@JonathanGiles - [ErrorCodeException](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/metricsadvisor/azure-ai-metricsadvisor/src/main/java/com/azure/ai/metricsadvisor/models/ErrorCodeException.java) is an exception type defined in swagger, it's done so because the error value from the MA service has a specific shape (shape represented by a complex type [ErrorCode](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/metricsadvisor/azure-ai-metricsadvisor/src/main/java/com/azure/ai/metricsadvisor/models/ErrorCode.java)).

Sameeksha and I are wondering what is the best approach here:- 

1. Rename `ErrorCodeException` to a name more specific to the service
2. Convert exception to HttpResponseException

1 means, we will end up having a service-specific exception type in the MA library. 2 means we will lose the type info and it make user inspection of error-value hard.

Reply:
Subclass the appropriate type in azure-core and provide the error code information through that subclass.